File: 프로그래머스/0/120887. k의 개수/k의 개수.java
class Solution {
    public int solution(int i, int j, int k) {
        int count = 0;

        for(int x = i; x <= j; x++) {
            String numString = String.valueOf(x);
            for(int y = 0; y < numString.length(); y++) {
                char c = numString.charAt(y);

                if(c == Character.forDigit(k, 10)) {
                    count++;
                }
            }
        }
        
        return count;
    }
}

File: 프로그래머스/0/120887. k의 개수/k의 개수.go
func solution(i int, j int, k int) int {
    count := 0
    
    for x := i; x <= j; x++ {
        num := x
        
        for num > 0 {
            if num % 10 == k {
                count++
            }
            num /= 10
        }
        
        if x == 0 && k == 0 {
            count++
        }
    }
    
    return count
}